pkg/operator/ceph/cluster/osd/osd_test.go

```diff
@@ -514,3 +528,395 @@ func TestGetOSDInfoWithCustomRoot(t *testing.T) {
 	assert.Equal(t, 0, len(osds3))
 	assert.NotNil(t, err)
 }
+
+func TestOSDsOnPVC(t *testing.T) {
+	logger.SetLevel(capnslog.TRACE) // want more log info for this test if it fails
+
+	namespace := "osd-on-pvc"
+	ctx := context.TODO()
+	// we don't need to create actual nodes for this test, but this is the set of nodes which should
+	// we will use to create fake placements for OSD prepare job pods
+	nodeList := []string{"node1", "node2", "node3"}
+	pickNode := func() string {
+		// simply return the node names from the list in a round-robin fashion
+		node := nodeList[0]
+		nodeList = append(nodeList[1:], node)
+		return node
+	}
+
+	// set up a fake k8s client set and watcher to generate events that the operator will listen to
+	clientset := fake.NewSimpleClientset()
+	d := clientset.Discovery()
+	fd, ok := d.(*fakediscovery.FakeDiscovery)
+	assert.True(t, ok)
+	// Set a fake k8s server version
+	fd.FakedServerVersion =
+		&version.Info{
+			Major:      "1",
+			Minor:      "13",
+			GitVersion: "v1.13.2",
+		}
+
+	os.Setenv(k8sutil.PodNamespaceEnvVar, namespace)
+	defer os.Unsetenv(k8sutil.PodNamespaceEnvVar)
+
+	var pvcReactor k8stesting.ReactionFunc = func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		// PVCs are created with generateName used, and we need to capture the create calls and
+		// generate a name for them in order for PVCs to all have unique names.
+		createAction, ok := action.(k8stesting.CreateAction)
+		if !ok {
+			t.Fatal("err! action is not a create action")
+			return false, nil, nil
+		}
+		obj := createAction.GetObject()
+		pvc, ok := obj.(*corev1.PersistentVolumeClaim)
+		if !ok {
+			t.Fatal("err! action not a PVC")
+			return false, nil, nil
+		}
+		if pvc.Name == "" {
+			// generate a uuid to add a random postfix to generateName
+			b := [16]byte(uuid.New())
+			// use base32 encoding to create a shorter uuid
+			b32 := base32.StdEncoding.EncodeToString(b[:])       // includes trailing equal signs
+			pvc.Name = pvc.GenerateName + strings.Trim(b32, "=") // trim off the trailing equal signs
```

Comment:
That looks good but feels unnecessary if we are not looping, can't we just have a hardcoded UUID, we re-use each time.?

Reply:
Travis' PR to allow PVCs to be removed removes the need for generating a name entirely. Since I can't make code progress on this right now, I'll probably just remove this when I have to fix merge conflicts rebasing on top of Travis' changes.